subsys/bluetooth/host/gatt.c

```diff
@@ -1527,11 +1679,16 @@ static u8_t notify_cb(const struct bt_gatt_attr *attr, void *user_data)
 
 		conn = bt_conn_lookup_addr_le(cfg->id, &cfg->peer);
 		if (!conn) {
-#if defined(CONFIG_BT_GATT_DYNAMIC_DB)
+#if defined(CONFIG_BT_GATT_SERVICE_CHANGED)
 			if (ccc->cfg_changed == sc_ccc_cfg_changed) {
-				sc_save(cfg, data->ind_params);
+				struct sc_data *sc;
+
+				sc = (struct sc_data *)data->ind_params->data;
+				sc_save(cfg->id, &cfg->peer,
+					sys_le16_to_cpu(sc->start),
```

Comment:
So this is suppose to store in LE format always? Afaik we use host byte order for settings.

Reply:
We actually store in host byte order, that is why this line converts the data from LE to CPU before calling sc_save.

Reply:
Hmm but then why the ind_params->data is carrying that in the over the air order? unpacked structs should always be in host byte order to avoid confusion.

Reply:
Indeed, it seems that gatt_indicate should be the one to perform that sys_cpu_to_le16 right before calling gatt_send.
Not sure if I should change this in this PR as it would break the exposed API (bt_gatt_indicate) that currently expects le16 data as input.
Maybe this should be a github issue or another PR, what do you say?

Reply:
Right, well I guess we should mark sc_data as packed then since it is indeed the data format to be sent over the air. Note that bt_gatt_indicate data byte order is not defined by the spec, this is up to the profile/service to define so we can't make any assumptions on byte order.

Reply:
> Well perhaps it not that easy to split the changes now that they are put together but logically there are independent changes, there is a dependency on the service changed Kconfig by the SC data persistency but that is about it. The advantage here is that the history is cleaner, we can bysect the changes independently, for instance you can have a clear explanation why BT_GATT_SERVICE_CHANGED may be needed even with dynamic db support disabled.

Makes sense, I'll give it a go then!

Reply:
Yes that must be the reason it was like that! Added __packed to sc_data in the latest push.